lib/ObjWriter/objwriter.cpp

```diff
@@ -13,838 +13,715 @@
-  const Target *TheTarget;
-};
-
-bool ObjectWriter::init(llvm::StringRef ObjectFilePath) {
-  llvm_shutdown_obj Y; // Call llvm_shutdown() on exit.
-
-  // Initialize targets
-  InitializeNativeTarget();
-  InitializeNativeTargetAsmPrinter();
-
-  MCOptions = InitMCTargetOptionsFromFlags();
-  TripleName = Triple::normalize(TripleName);
-
-  TheTarget = GetTarget();
-  if (!TheTarget)
-    return error("Unable to get Target");
-  // Now that GetTarget() has (potentially) replaced TripleName, it's safe to
-  // construct the Triple object.
-  Triple TheTriple(TripleName);
-
-  std::error_code EC;
-  OS.reset(new raw_fd_ostream(ObjectFilePath, EC, sys::fs::F_None));
-  if (EC)
-    return error("Unable to create file for " + ObjectFilePath + ": " +
-                 EC.message());
-
-  MRI.reset(TheTarget->createMCRegInfo(TripleName));
-  if (!MRI)
-    return error("Unable to create target register info!");
-
-  MAI.reset(TheTarget->createMCAsmInfo(*MRI, TripleName));
-  if (!MAI)
-    return error("Unable to create target asm info!");
-
-  MOFI.reset(new MCObjectFileInfo);
-  MC.reset(new MCContext(MAI.get(), MRI.get(), MOFI.get()));
-  MOFI->InitMCObjectFileInfo(TheTriple, RelocModel, CMModel, *MC);
-
-  std::string FeaturesStr;
-
-  MII.reset(TheTarget->createMCInstrInfo());
-  if (!MII)
-    return error("no instr info info for target " + TripleName);
-
-  MSTI.reset(TheTarget->createMCSubtargetInfo(TripleName, MCPU, FeaturesStr));
-  if (!MSTI)
-    return error("no subtarget info for target " + TripleName);
-
-  MCE = TheTarget->createMCCodeEmitter(*MII, *MRI, *MC);
-  if (!MCE)
-    return error("no code emitter for target " + TripleName);
-
-  MAB = TheTarget->createMCAsmBackend(*MRI, TripleName, MCPU, MCOptions);
-  if (!MAB)
-    return error("no asm backend for target " + TripleName);
-
-  MS = TheTarget->createMCObjectStreamer(TheTriple, *MC, *MAB, *OS, MCE, *MSTI,
-                                         RelaxAll,
-                                         /*IncrementalLinkerCompatible*/ true,
-                                         /*DWARFMustBeAtTheEnd*/ false);
-  if (!MS)
-    return error("no object streamer for target " + TripleName);
-
-  TM.reset(TheTarget->createTargetMachine(TripleName, MCPU, FeaturesStr,
-                                          TargetOptions(), None));
-  if (!TM)
-    return error("no target machine for target " + TripleName);
-
-  Asm.reset(TheTarget->createAsmPrinter(*TM, std::unique_ptr<MCStreamer>(MS)));
-  if (!Asm)
-    return error("no asm printer for target " + TripleName);
-
-  FrameOpened = false;
-  FuncId = 1;
-
-  return true;
+Triple ObjectWriter::GetTriple()
+{
+	Triple TheTriple(TripleName);
+
+	if (TheTriple.getOS() == Triple::OSType::Darwin)
+	{
+		TheTriple = Triple(TheTriple.getArchName(), TheTriple.getVendorName(), "darwin",
+			TheTriple.getEnvironmentName()); // it is workaround for llvm bug
+											 // https://bugs.llvm.org//show_bug.cgi?id=24927.
```

Comment:
Spacing is weird - I notice you switched to tabs. The code-gen team was following the [LLVM coding guidelines](http://llvm.org/docs/CodingStandards.html) for LLILC (including running clang-format and clang-tidy during CI) which is why this file was formatted weirdly. I would preserve the existing style personally, but that's up to your team.